cmds/core/wget/wget.go

```diff
@@ -32,42 +32,41 @@ import (
 	"net/url"
 	"os"
 	"path"
+	"strings"
 
 	"github.com/u-root/u-root/pkg/curl"
 	"github.com/u-root/u-root/pkg/uio"
 )
 
 var outPath = flag.String("O", "", "output file")
+var errEmptyURL = errors.New("empty url")
 
-func usage() {
-	log.Printf("Usage: %s [ARGS] URL\n", os.Args[0])
-	flag.PrintDefaults()
-	os.Exit(2)
+type command struct {
+	url        string
+	outputPath string
 }
 
-func run() (reterr error) {
-	log.SetPrefix("wget: ")
-
-	if flag.Parse(); flag.NArg() != 1 {
-		usage()
+func New(outPath string, url string) *command {
```

Comment:
this can be new, not New, otherwise some of the code checkers will complain, and we don't need to export it.

Reply:
Yeah sure, but I make it New because otherwise code checkers may or will complain for shadowing builtin function, this is not an issue here, right now and probably never will, I don't expect someone will want to call builtin. That why I initially named it `tee`, but probably `newCommand` or `newUtil` will be a compromise?